server/src/database/delayedStorage.ts

```diff
@@ -0,0 +1,62 @@
+export class DelayedStorage<T> {
```

Comment:
I don't know that I am convinced this abstracton adds any value to our use case.  I feel like having 2 functions:
`backupGame(gameId: string, game: BaseUnifiedGame): Promise<void>`
and:
`checkForBackup(gameId: string): Promise<BaseUnifiedGame | null>`
will do what we need.

MAybe we need to batch in the future, but writing to S3 won't be expensive, so I don't think we need to create arbitrary delays.